user_sync/connector/directory_ldap.py

```diff
@@ -269,35 +272,55 @@ def iter_users(self, users_filter):
                 if (last_attribute_name != None):
                     self.logger.warn('No email attribute: %s for dn: %s', last_attribute_name, dn)
                 continue
+
+            source_attributes = {}
             
             user = user_sync.connector.helper.create_blank_user()
             user['email'] = email
+            source_attributes['email'] = email
                 
             username, last_attribute_name = self.user_username_formatter.generate_value(record)
             if (username == None and last_attribute_name != None):
                 self.logger.info('No username attribute: %s for dn: %s', last_attribute_name, dn)    
             user['username'] = username if username != None else email
+            if (username != None):
+                source_attributes['username'] = username
                     
             domain, last_attribute_name = self.user_domain_formatter.generate_value(record)
             if (domain != None):
                 user['domain'] = domain
+                source_attributes['domain'] = domain
             elif (last_attribute_name != None):
                 self.logger.info('No domain attribute: %s for dn: %s', last_attribute_name, dn)    
                                                 
             given_name_value = LDAPValueFormatter.get_attribute_value(record, 'givenName')
             if (given_name_value != None):   
                 user['firstname'] = given_name_value
+                source_attributes['givenName'] = given_name_value
```

Comment:
Why not make the keys of `source_attributes` match the keys of the source customer record (i.e., `firstname` and `lastname` and `country`) as constructed here?  I don't think we want to force the customer to know which particular alias for `gn` and `sn` we are using in our code.  The actual source attribute in the directory could be any one of the LDAP aliases for `gn` and `sn`, so it's not like we are guaranteed to be using the actual value in the source directory. So as long as we are picking `firstname` and `lastname` as the canonical values we use on the Adobe side I think we should just use them on the customer side in the extensions as well.  After all, if the extension wants to set those value it's going to have to know those names.